### PR TITLE
Rework API fixtures to be type safe

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-12132
affects: none
release note: none

### CODE REVIEW

**Comment by vanwagonet** [comment]
defining the `make` func is more code than the templates, but then each use is less typing, safe, auto-completable, & you can jump to definition to see the defaults.

File: Core/CoreTests/Database/Models/PermissionsTests.swift [hunk 1]
```diff
@@ -91,22 +91,22 @@ class PermissionsTests: CoreTestCase {
...
+            view_error_reports: true, view_grade_changes: true, view_jobs: true, view_notifications: true, view_statistics: true,
+            undelete_courses: true, change_course_state: true, comment_on_others_submissions: true, create_collaborations: true,
+            create_conferences: true, create_forum: true, generate_observer_pairing_code: true, import_outcomes: true, lti_add_edit: true,
+            manage_admin_users: true, manage_assignments: true, manage_calendar: true, manage_content: true, manage_files: true,
+            manage_grades: true, manage_groups: true, manage_interaction_alerts: true, manage_outcomes: true, manage_sections: true,
+            manage_students: true, manage_user_notes: true, manage_rubrics: true, manage_wiki: true, moderate_forum: true,
+            post_to_forum: true, read_announcements: true, read_email_addresses: true, read_forum: true, read_question_banks: true,
+            read_reports: true, read_roster: true, read_sis: true, select_final_grade: true, send_messages: true, send_messages_all: true,
+            view_all_grades: true, view_audit_trail: true, view_group_pages: true, view_user_logins: true
+        )
```
Comment (shichongrui) [inline Core/CoreTests/Database/Models/PermissionsTests.swift]: woah

File: TestsFoundation/TestsFoundation/Fixtures/API/APIPermissions.swift [hunk 2]
```diff
@@ -17,8 +17,146 @@
...
+            read_roster: read_roster,
+            read_sis: read_sis,
+            select_final_grade: select_final_grade,
+            send_messages: send_messages,
+            send_messages_all: send_messages_all,
+            view_all_grades: view_all_grades,
+            view_audit_trail: view_audit_trail,
+            view_group_pages: view_group_pages,
+            view_user_logins: view_user_logins
+        )
```
Comment (shichongrui) [inline TestsFoundation/TestsFoundation/Fixtures/API/APIPermissions.swift]: double woah

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-12132](https://instructure.atlassian.net/browse/MBL-12132)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 94.96% | 95.04% | -0.08%
**Student** | 92.34% | 92.34% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.71% | 92.71% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/CoreTests/Database/Models/PermissionsTests.swift (hunk 1 above)
Comment (armstrongnate) [reply Core/CoreTests/Database/Models/PermissionsTests.swift]: Wow!

File: TestsFoundation/TestsFoundation/Fixtures/API/APIPermissions.swift (hunk 2 above)
Comment (armstrongnate) [reply TestsFoundation/TestsFoundation/Fixtures/API/APIPermissions.swift]: lol

**Pull request merged**